File: csharp/sprint5/C/Solution.cs
/*
// закомментируйте перед отправкой
public class Node
{
    public int Value { get; set;}
    public Node Left { get; set;}
    public Node Right { get; set;}

    public Node(int value)
    {
        Value = value;
        Left = null;
        Right = null;
    }
}
*/

public class Solution
{
    public static bool Solve(Node root)
    {
        // Your code
        // “ヽ(´▽｀)ノ”
    }

    private static void Test()
    {
        var node1 = new Node(3);
        var node2 = new Node(4);
        var node3 = new Node(4);
        var node4 = new Node(3);
        var node5 = new Node(2)
        {
            Left = node1,
            Right = node2,
        };

        var node6 = new Node(2)
        {
            Left = node3,
            Right = node4
        };

        var node7 = new Node(1)
        {
            Left = node5,
            Right = node6
        };

        Console.WriteLine(Solve(node7));
    }
}

File: swift/sprint5/C/solution.swift
/*
 Comment it before submitting

 open class Node {
     
     public var value: Int
     public var left: Node?
     public var right: Node?
     
     public init(_ value: Int) {
         self.value = value
         self.right = nil
         self.left = nil
     }
 }
 */

func solution(node: Node?) -> Bool {
    // Your code
    // ヽ(´▽`)/
    return false
}


func test() {
    var node1 = Node(3)
    var node2 = Node(4)
    var node3 = Node(4)
    var node4 = Node(3)
    var node5 = Node(2)
    node5.left = node1
    node5.right = node2
    var node6 = Node(2)
    node6.left = node3
    node6.right = node4
    var node7 = Node(1)
    node7.left = node5
    node7.right = node6
    assert(solution(node: node7))
}
